src/enumerable.cr

```diff
@@ -303,8 +303,16 @@ module Enumerable(T)
     raise ArgumentError.new("size must be positive") if size <= 0
 
     each_slice(size) do |slice|
-      group = slice.size < size ? slice + [filled_up_with] : slice
-      yield group
+      if slice.size < size
+        paddings = Array(T | U).new
+        (size - slice.size).times do
+          paddings << filled_up_with
+        end
+        group = slice + paddings
+        yield group
+      else
+        yield slice
+      end
```

Comment:
How about just
```cr
(size - slice.size).times { slice << filled_up_with }
yield slice
```

Or perhaps 
```cr
slice.concat Array(T|U).new(size - slice.size, filled_up_with)
yield slice
```

? It would be interesting to see what's more expensive (slower), potentially repeated resizings of pushing n-m times or allocating a whole new array. Actually does `Array#fill` resize? I can't remember.